code/_core/obj/structure/totem.dm

```diff
@@ -0,0 +1,243 @@
+/obj/structure/totem
+	name = "totem"
+	desc = "None."
+	desc_extended = "A totem that will produce a buff for the caster and their allies."
+	icon = 'icons/obj/structure/totem.dmi'
+
+	var/next_fire = 0
+	var/cooldown_fire = SECONDS_TO_DECISECONDS(1)
+
+	var/mob/living/owner //the object's owner
+	var/totem_remove_time = 0
+
+	var/leveled_effect = 0
+
+	var/affecting_faction //which faction it should affect
+
+/obj/structure/totem/Finalize()
+	. = ..()
+	start_thinking(src)
+
+/obj/structure/totem/Destroy()
+	if(owner)
+		owner.totem = null
+	return ..()
+
+/obj/structure/totem/think()
+	if(world.time >= totem_remove_time)
+		Destroy()
+		return FALSE
+	if(world.time <= next_fire)
+		return TRUE
+	next_fire = world.time + cooldown_fire
+	totemic_effect()
+	return TRUE
+
+/obj/structure/totem/proc/totemic_effect()
+	return
+
+/*******************
+*******TOTEMS*******
+*******************/
+
+/obj/structure/totem/health_heal
+	name = "totem of health regeneration"
+	desc_extended = "A totem that will restore the caster's and their allies' health."
+	icon_state = "health"
+
+/obj/structure/totem/health_heal/totemic_effect() //copy paste from staff of healing's think, with slight modifications
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag != affecting_faction) //!= because we want to only affect allies
+			continue
+		if(!istype(L.health))
+			continue
+		if(L.health.health_current >= L.health.health_max)
+			continue
+		if(L.health.get_loss(BRUTE))
+			L.brute_regen_buffer += (3 + (3 * leveled_effect))
+		if(L.health.get_loss(BURN))
+			L.burn_regen_buffer += (3 + (3 * leveled_effect))
+		if(L.health.get_loss(TOX))
+			L.tox_regen_buffer += (3 + (3 * leveled_effect))
+		var/obj/effect/temp/healing/H = new(L.loc,10,COLOR_RED)
```

Comment:
You should use ``CREATE(/obj/effect/temp/healing/,get_turf(L))`` instead and remove the ``INITIALIZE(H)`` and friends.